source/beamline/BioXAS/BioXASSideTableCoordinator.cpp

```diff
@@ -371,16 +360,42 @@ void BioXASSideTableCoordinator::onYawStopControlValueChanged()
 	}
 }
 
-void BioXASSideTableCoordinator::manipulateVerticalPVChange(double upstreamHeight, double downstreamHeight, AMControl* heightPV, AMControl *pitchPV)
+double BioXASSideTableCoordinator::calculateTableHeight(const double upstreamHeight, const double downstreamHeight) const
 {
 	// The vertical height is the height of the center of the table, which is the average of upstream value and the downstream value
-	double height = (upstreamHeight + downstreamHeight) / 2;
+	return (upstreamHeight + downstreamHeight) / 2;
+}
+
+double BioXASSideTableCoordinator::calculateTablePitch(const double upstreamHeight, const double downstreamHeight) const
+{
 	// The pitch is the degree of the arcTangent value of (dH - uH) / distance
-	double pitch =  radianToDegree(atan((downstreamHeight - upstreamHeight) / (tableVerticalMotorPosition_ * 2)));
+	return radianToDegree(atan((downstreamHeight - upstreamHeight) / (tableVerticalMotorPosition_ * 2)));
+}
+
+double BioXASSideTableCoordinator::calculateTableLateral(const double upstreamLateral, const double downstreamLateral) const
+{
+	// The lateral is the lateral of the center of the table, which is the average of upstream value and the downstream value
+	return ( upstreamLateral + downstreamLateral ) / 2;
+}
+
+double BioXASSideTableCoordinator::calculateTableYaw(const double upstreamLateral, const double downstreamLateral) const
+{
+	// The yaw is the degree of the arcTangent value of (dH - uH) / distance
+	return radianToDegree(atan((downstreamLateral - upstreamLateral) / (tableHorizontalMotorPosition_ * 2)));
+}
+
+void BioXASSideTableCoordinator::manipulateVerticalPVChange(double upstreamHeight, double downstreamHeight, AMControl* heightPV, AMControl *pitchPV)
+{
+	double height = calculateTableHeight(upstreamHeight, downstreamHeight);
+	double pitch =  calculateTablePitch(upstreamHeight, downstreamHeight);
+
+	height = static_cast<double>(static_cast<int>(height*1000+0.5))/1000.0;
+	pitch = static_cast<double>(static_cast<int>(pitch*1000+0.5))/1000.0;
 
 	if (!heightPV->withinTolerance(height)) {
```

Comment:
Have you tested these motors in a scan (such as the generic scanner)?  I have a feeling code like this if statement will cause some issues because controls are still expected to go through it's move status state machine even if it isn't going anywhere.